core/scripts/keeper/main.go

```diff
@@ -0,0 +1,287 @@
+		common.HexToAddress(config.LinkTokenAddr),
+		common.HexToAddress(config.LinkETHFeedAddr),
+		common.HexToAddress(config.FastGasFeedAddr),
+		config.PaymentPremiumPBB,
+		config.FlatFeeMicroLink,
+		big.NewInt(config.BlockCountPerTurn),
+		config.CheckGasLimit,
+		big.NewInt(config.StalenessSeconds),
+		config.GasCeilingMultiplier,
+		big.NewInt(config.FallbackGasPrice),
+		big.NewInt(config.FallbackLinkPrice),
+	)
+	if err != nil {
+		log.Fatal("DeployAbi failed: ", err)
+	}
+	waitForTx(deployKeeperRegistryTx.Hash())
+	log.Println("KeeperRegistry deployed:", registryAddr.Hex(), "-", deployKeeperRegistryTx.Hash().Hex())
+
+	// Approve keeper registry
+	approveRegistryTx, err := linkToken.Approve(buildTxOpts(), registryAddr, approveAmount)
+	if err != nil {
+		log.Fatal("Approve failed: ", err)
+	}
+	waitForTx(approveRegistryTx.Hash())
+	log.Println("KeeperRegistry approved:", registryAddr.Hex(), "-", approveRegistryTx.Hash().Hex())
+
+	// Deploy Upkeeps
+	deployUpkeeps(registryInstance)
+
+	// Set Keepers
+	keepers, owners := config.keepers()
+	setKeepersTx, err := registryInstance.SetKeepers(buildTxOpts(), keepers, owners)
+	if err != nil {
+		log.Fatal("SetKeepers failed: ", err)
+	}
+	waitForTx(setKeepersTx.Hash())
+	log.Println("Keepers registered:", setKeepersTx.Hash().Hex())
+}
+
+func deployUpkeeps(registryInstance *keeper.KeeperRegistry) {
+	for i := int64(0); i < config.UpkeepCount; i++ {
+		fmt.Println()
+
+		// Deploy
+		upkeepAddr, deployUpkeepTx, _, err := upkeep.DeployUpkeepPerformCounterRestrictive(buildTxOpts(), client,
+			big.NewInt(config.UpkeepTestRange), big.NewInt(config.UpkeepAverageEligibilityCadence),
+		)
+		if err != nil {
+			log.Fatal(i, "- DeployAbi failed: ", err)
+		}
+		waitForTx(deployUpkeepTx.Hash())
+		log.Println(i, "- Upkeep deployed:", upkeepAddr.Hex(), "-", deployUpkeepTx.Hash().Hex())
+
+		// Approve
+		approveUpkeepTx, err := linkToken.Approve(buildTxOpts(), upkeepAddr, approveAmount)
+		if err != nil {
+			log.Fatal(i, "- Approve failed: ", err)
+		}
+		waitForTx(approveUpkeepTx.Hash())
+		log.Println(i, "- Upkeep approved:", upkeepAddr.Hex(), "-", approveUpkeepTx.Hash().Hex())
+
+		// Register
+		registerUpkeepTx, err := registryInstance.RegisterUpkeep(buildTxOpts(),
+			upkeepAddr, config.UpkeepGasLimit, fromAddr, []byte(config.UpkeepCheckData),
+		)
+		if err != nil {
+			log.Fatal(i, "- RegisterUpkeep failed: ", err)
+		}
+		waitForTx(registerUpkeepTx.Hash())
+		log.Println(i, "- Upkeep registered:", upkeepAddr.Hex(), "-", registerUpkeepTx.Hash().Hex())
+
+		// Fund
+		addFundsTx, err := registryInstance.AddFunds(buildTxOpts(), big.NewInt(int64(i)), addFundsAmount)
+		if err != nil {
+			log.Fatal(i, "- AddFunds failed: ", err)
+		}
+		waitForTx(addFundsTx.Hash())
+		log.Println(i, "- Upkeep funded:", upkeepAddr.Hex(), "-", addFundsTx.Hash().Hex())
+	}
+	fmt.Println()
+}
+
+func waitForTx(tx common.Hash) int {
```

Comment:
I think you can use `bind.WaitMined` or `bind.WaitDeployed` depending on whether tx is a regular one or a contract deployment tx.

Reply:
Good point, done